contrib/utilities/indent-branch

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2018 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+#
+# This script does the same thing as contrib/utilities/indent but only reformats
+# files which have changed (or been added and not committed) since the last
+# merge commit to the master branch. It should be run as
+#
+#    make indent-branch
+#
+# from a build directory.
+#
+
+if test ! -d source -o ! -d include -o ! -d examples ; then
+  echo "*** This script must be run from the top-level directory of deal.II."
+  exit 1
+fi
+
+# check that clang-format is available:
+source contrib/utilities/indent-checks
+
+# Filenames may (in theory) may consist of any characters besides '/' and NUL
+# (including newlines, terminal commands, etc). Hence, this script uses NUL as
+# the file name separation character. Since bash cannot store NUL-delimited
+# output in variables we convert it back-and-forth from base64.
+
+# add any new files:
+UNSORTED_FILES=$(git ls-files -z --others --exclude-standard | base64)
+# add any files changed relative to the last merge commit to master:
+LAST_MERGE_COMMIT=$(git log --format="%H" --merges --max-count=1 master)
```

Comment:
Can you select which merges count? Here, you're getting all of them, including where someone happens to merge from mainline, or merges a pull request someone else may have had against the current branch. What we really want is the last pull request merge that was merged via github, if that is even possible to identify.

In practice, what you have here is almost certainly good enough. We'll catch all of the other cases through the CI checker, and can then advise the submitter to run `make indent-all` or something. So if the idea above is not feasible, don't worry.

Reply:
I don't think its possible to identify merges done on GitHub. They have the generic message

    merged pull request #9999 from account/branch

Its possible that someone could merge things in a strange way on master and get a similar message; I don't think that there is a good way to derive whether or not something happened with the master branch on github from just local information.